docs/spec/reactors/consensus/proposer-selection.md

```diff
@@ -2,20 +2,228 @@
+        for each validator i in vset:
+            A(i) += VP(i)
+        prop = max(A)
+        A(prop) -= P
+    
+    def proposalSelection (vset, times):
+        for in in range(times):
+        singleProposalSelection(vset)
+
+        // normalize - shift with average priority
+        avg = sum(A(i) for i in vset)/len(vset)
+        for each validator i in vset:
+            A(i) -= avg
+
+Note that the shifting operation happens after singleProposalSelection() has run times times.
+
+#### New Validator
+When a new validator is added same problem as the one described for removal appears, the sum of priorities in the new set is not zero. This is now fixed with the shift step introduced above.
+
+One other issue that needs to be addressed is the following. A validator V with low voting power that has just been elected is moved to the end of the queue. If the validator set is large and/ or with other validators with significantly higher power, validator V will have to wait a lot of rounds to be elected. If V could remove and read itself to the validator set, it would make a significant (albeit unfair) "jump" ahead in the queue, decreasing the numbers of rounds it has to wait. 
+
+In order to prevent this, when a new validator is added, its initial priority is not set to 0 but to a lower value. This value is determined from the simulation of the above scenario: it is assumed that V had just proposed a block and was at the back of the queue when it was removed and re-added. In addition, a penalty factor is applied.
+
+If `vset = {v1,..,vn, V}` and `V` would be selected as proposer then it would have gone through the following changes:
+
+    ...
+      A(V)+=VP(V) // part of first for loop
+    ...
+    A(V)-=P  
+
+It follows that `A(V)-=sum(VP(i) for i in {v1,..,vn})`
+
+Therefore when `V` is added to the `{v1,..,vn}` set its initial priority will be:
+
+    A(V) = -sum(VP(i) for i in {v1,..,vn})
+
+With a penalty factor of 1.125, the initial priority for a newly added verifier is:
+    
+    A(V) = -1.125 * P
+
+If we consider the validator set where p3 has just been added:
+
+Validator | p1 | p2 | p3
+----------|--- |--- |---
+VP        | 1  | 3  | 8
+
+Let's assume that the last round was R and the proposer priorities were as shown in first row (R end) with their sum being 0. p3 is added with A(p3) = -4
+In the next round, p1 will still be ahead in the queue, elected as proposer and move back in the queue (same position it was just added)
+
+|Priority Round  | -4 | -3 | -2 | -1 | 0  | 1 | 2 | 3 | 4 | Alg step
+|--------------- | ---|--- |----|--- |--- |---|---|---|---|--------
+|R end          |    |    | p2 |    |    |   | p1|   |   |add p3
+|                | p3 |    | p2 |    |    |   | p1|   |   |A(p3) = -4
+|1              |    |    |    |    |    | p2|   | p1| p3|A(i)+=VP(i)
+|               | p3 |    |    |    |    | p2|   | p1|   |A(p3)-=P
+
+
+### Wrinkles
+
+#### Validator Power Overflow Conditions
```

Comment:
I thought we were normalizing the priorities? That would help prevent overflow, but if I recall correctly, it was also necessary for the @cwgoes proof

Reply:
This is for the voting power.